src/Storages/ObjectStorageQueue/ObjectStorageQueueIFileMetadata.cpp

```diff
@@ -260,13 +264,74 @@ void ObjectStorageQueueIFileMetadata::resetProcessing()
     if (state != FileStatus::State::Processing)
         throw Exception(ErrorCodes::LOGICAL_ERROR, "Cannot reset non-processing state: {}", state);
 
-    resetProcessingImpl();
-    file_status->reset();
-}
+    SCOPE_EXIT({
+        file_status->reset();
+    });
 
-void ObjectStorageQueueIFileMetadata::resetProcessingImpl()
-{
-    throw Exception(ErrorCodes::LOGICAL_ERROR, "resetProcessingImpl is not implemented");
+    if (!processing_id_version.has_value())
+    {
+        throw Exception(
+            ErrorCodes::LOGICAL_ERROR,
+            "No processing id version set, but state is `Processing` ({})",
+            node_metadata.toString());
+    }
+
+    size_t check_processing_id_path_idx = 0;
+    size_t remove_processing_id_path_idx = 1;
+    size_t remove_processing_path_idx = 2;
+
+    Coordination::Requests requests;
+    requests.push_back(zkutil::makeCheckRequest(processing_node_id_path, processing_id_version.value()));
```

Comment:
why do you need this check? Remove will fail if it doesn't exist

Reply:
Yes, it needs to be removed